pkg/pool-weighted/contracts/WeightedMath.sol

```diff
@@ -73,6 +73,20 @@ library WeightedMath {
         _require(invariant > 0, Errors.ZERO_INVARIANT);
     }
 
+    // Computes a weighted product (same math as invariant) while ignoring ones.
+    function _calculateWeightedProduct(uint256[] memory normalizedWeights, uint256[] memory balances)
```

Comment:
Was it a deliberate decision to call the 2nd argument `balances`? If so, why? If not, can we change this to `rates`?

Reply:
Although it's only used in one place for now, this is meant to be a generic function. That's why it's called `_calculateWeightedProduct` instead of, for example, `_calculateYield`.

But being generic, `balances` also no longer makes sense (it was blindly copied from the signature of `_calculateInvariant`). What's a good generic name?

Reply:
`amounts`